src/zjs_ipm.h

```diff
@@ -95,6 +124,15 @@ typedef struct zjs_ipm_message {
             uint8_t color_b;
             void *buffer;
         } glcd;
+#endif
+
+#ifdef BUILD_MODULE_SENSOR
+        struct sensor_data {
+            enum sensor_channel channel;
+            uint32_t frequency;
+            double value[3];
```

Comment:
Would you prefer it to split this struct into 2 ? Like we have for SensorReading (frequency&timestamp) as base struct, and say concreteSensor's reading (which contains the actual value) like https://w3c.github.io/gyroscope/#gyroscopereading. 
Also no timestamp ? like performance.now() equivalent .

When you implement the next sensor, say AmbientLightSensor , there is only 1 double for "value".
If you implement Uncalibrated Magnetometer or Uncalibrated Gyro, then double value[6].